pkg/types/gcp/validation/platform.go

```diff
@@ -69,7 +70,16 @@ func ValidatePlatform(p *gcp.Platform, fldPath *field.Path) field.ErrorList {
 		allErrs = append(allErrs, ValidateMachinePool(p, p.DefaultMachinePlatform, fldPath.Child("defaultMachinePlatform"))...)
 		allErrs = append(allErrs, ValidateDefaultDiskType(p.DefaultMachinePlatform, fldPath.Child("defaultMachinePlatform"))...)
 	}
-	if p.Network != "" {
+	if p.NetworkProjectID != "" {
+		if p.Network == "" {
+			allErrs = append(allErrs, field.Required(fldPath.Child("network"), "must provide a network when a network id is specified"))
+		}
+		if ic.CredentialsMode != types.ManualCredentialsMode && ic.CredentialsMode != types.PassthroughCredentialsMode {
+			allErrs = append(allErrs, field.NotSupported(fldPath.Child("credentialsMode"),
+				ic.CredentialsMode, []string{string(types.ManualCredentialsMode), string(types.PassthroughCredentialsMode)}))
+		}
+	}
+	if p.Network != "" || p.NetworkProjectID != "" {
```

Comment:
I think we can just keep this condition as `p.Network != ""`. If users just specified p.NetworkProjectID, I think it's ok that these error messages would get skipped. I also suggest an extra line also:
```suggestion
	
	if p.Network != ""  {
```